### PR TITLE
Update cdn for new format

### PR DESCRIPTION
**Pull request by murderteeth**
Updates the CDN API to support the new consolidated vault structure introduced in `refactor-vaults`, where all vaults for a chain are stored in a single JSON file (e.g., `vaults/1.json`) instead of individual files per vault.

## Changes

**Dual Access Pattern Support:**
- **New format**: Fetch entire chain file directly
  - `GET /api/cdn/vaults/1.json` → Returns all Ethereum vaults
  - Enables efficient bulk lookups and reduces HTTP requests

- **Backwards compatibility**: Single vault requests continue working
  - `GET /api/cdn/vaults/1/0xabc...123.json` → Returns individual vault
  - Automatically extracts from chain file using lowercase-normalized address
  - Returns 404 if vault not found
  - Both lowercase and checksum addresses will now be supported

**CDN Optimization:**
- Use commit SHA (`VERCEL_GIT_COMMIT_SHA`) instead of `master` branch reference
- Ensures immediate data freshness on deployments
- Bypasses jsDelivr's slow branch cache expiration (hours/days)
- Atomic deployments: code and data always in sync

**Automated tests:**
- Added comprehensive test suite for `getPath()` function
- Validates pattern detection, address normalization, and edge cases
- 13 tests covering both access patterns

## Manual testing
I tested these changes manually by fetching these different urls from this PRs preview link
https://risk-score-git-update-cdn-for-new-format-yearn.vercel.app/cdn/vaults/1.json

https://risk-score-git-update-cdn-for-new-format-yearn.vercel.app/cdn/vaults/1/0x074134a2784f4f66b6ced6f68849382990ff3215.json

https://risk-score-git-update-cdn-for-new-format-yearn.vercel.app/cdn/vaults/1/0x074134A2784F4F66b6ceD6f68849382990Ff3215.json

https://risk-score-git-update-cdn-for-new-format-yearn.vercel.app/cdn/asset/ETHplus.md

### CODE REVIEW

File: api/cdn.ts [hunk 1]
```diff
@@ -2,23 +2,43 @@ export const config = { runtime: 'edge' }
...
+
+  // Detect single vault pattern: vaults/{chainId}/{address}.json
+  const singleVaultMatch = path.match(/^vaults\/(\d+)\/(0x[a-fA-F0-9]{40})\.json$/)
+  if (singleVaultMatch) {
+    const [, chainId, address] = singleVaultMatch
+    return {
+      type: 'single-vault',
+      chainId,
+      address: address.toLowerCase(),
+      path: `vaults/${chainId}.json`,
```
[the remark this answers is not in the document]
Comment (murderteeth) [reply api/cdn.ts]: what's happening here is the path is actually being rewritten *from* something like `vaults/chainid/address.json` to `vaults/chainid.json`. because that is the actual file path now, vault/chainid/address.json no longer exists. later in the code chainid.json is parsed and the specific vault gets indexed from that, https://github.com/yearn/risk-score/blob/ff6afeca163ff5f6a2407ffec7e5031a0f10f999/api/cdn.ts#L87

[the remark this answers is not in the document]
Comment (w84april) [reply api/cdn.ts]: oh! It all came together, thanks
